nixos/modules/services/web-apps/peertube.nix

```diff
@@ -0,0 +1,186 @@
+{ lib, pkgs, config, ... }:
+
+let
+  name = "peertube";
+  cfg = config.services.peertube;
+
+  uid = config.ids.uids.peertube;
+  gid = config.ids.gids.peertube;
+in
+{
+  options.services.peertube = {
+    enable = lib.mkEnableOption "Enable Peertube’s service";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = name;
+      description = "User account under which Peertube runs";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = name;
+      description = "Group under which Peertube runs";
+    };
+
+    configFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        The configuration file path for Peertube.
+      '';
+    };
+
+    database = {
+      createLocally = lib.mkOption {
+        description = "Configure local PostgreSQL database server for PeerTube.";
+        type = lib.types.bool;
+        default = true;
+      };
+
+      name = lib.mkOption {
+        type = lib.types.str;
+        default = "peertube_prod";
+        description = "Database name.";
+      };
+
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "peertube";
+        description = "Database user.";
+      };
+    };
+
+    smtp = {
+      createLocally = lib.mkOption {
+        description = "Configure local Postfix SMTP server for PeerTube.";
+        type = lib.types.bool;
+        default = true;
+      };
+    };
+
+    redis = {
+      createLocally = lib.mkOption {
+        description = "Configure local Redis server for PeerTube.";
+        type = lib.types.bool;
+        default = true;
+      };
+    };
+
+    runtimeDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/${name}";
+      description = "The directory where Peertube stores its runtime data.";
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.peertube;
+      description = ''
+        Peertube package to use.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users = lib.optionalAttrs (cfg.user == name) {
+      "${name}" = {
+        inherit uid;
+        group = cfg.group;
+        description = "Peertube user";
+        home = cfg.runtimeDir;
+        useDefaultShell = true;
+        # todo: fix this. needed for postgres authentication
+        password = "peertube";
```

Comment:
This would need to be fixed I think. Does peertube not support unix socket authentication for postgresql as I think this is the approach other services take?

Reply:
True, it might do, but I didn't work that out yet. Yeah that's why I suggested maybe just getting the `pkg` in might be easier, since modules are always a bit more opinionated..

Reply:
I think I will look into this later.

Reply:
See https://github.com/stevenroose/nixpkgs/pull/2